lib/config/validation.ts

```diff
@@ -27,9 +27,19 @@ const options = getOptions();
 
 let optionTypes: Record<string, RenovateOptions['type']>;
 let optionParents: Record<string, RenovateOptions['parent']>;
+let optionGlobals: Record<string, RenovateOptions['globalOnly']>;
 
 const managerList = getManagerList();
 
+// allow these global options in repo config
+const allowedGlobalOptions = [
+  'customEnvVariables',
+  'migratePresets',
+  'productLinks',
+  'secrets',
+  'customizeDashboard',
+];
```

Comment:
@rarkins why do we have these excludes? (copied from code below) shouldn't we remove the global only flag from option instead if they are allowed in repo config?

Reply:
I don't understand why.

These are/were handled separately because they're _objects_, but this doesn't mean all of them are allowed in repo config

Reply:
We should keep it for now, and I will remove them in a new PR when new validation is added to validate the config options.